src/lib/check-in.ts

```diff
@@ -18,45 +89,35 @@ export function makeFetchCheckinDataAttempts(
     };
   };
 
-  return SwClient.loadJsonPage<Response>({
-    url: SwClient.withSuffix(
-      'mobile-air-operations/v1/mobile-air-operations/page/check-in/',
-      reservation
-    ),
-    method: SwClient.Method.GET,
-    headers,
-    retry: {
-      // We'll try a total of 80 times.
-      // This should give us 5 seconds of checkin tries before checkin time and 15 seconds after.
-      limit: 79,
-      methods: [SwClient.Method.POST],
-      statusCodes: [400, HttpStatus.NOT_FOUND],
-      calculateDelay: state => {
-        const nowTimestamp = Luxon.DateTime.now().toLocaleString(
-          Luxon.DateTime.DATETIME_FULL_WITH_SECONDS
-        );
-        if (logger) {
-          // It's normal to see many of these messages before we get the good response. We try
-          // many times because checkin does not become available at exactly the time expected.
-          logger.log(
-            'Failed on attempt %d of %d at %s with error:',
-            state.attemptCount,
-            state.retryOptions.limit,
-            nowTimestamp,
-            state.error.response?.body
-          );
-        }
-
-        // cancel retry when limit is hit
-        if (state.attemptCount > state.retryOptions.limit) {
-          return 0;
-        }
-
-        // retry every 250 milliseconds
-        return 250;
-      }
+  let response;
+  try {
+    response = await SwClient.loadJsonPage<Response>({
+      url: SwClient.withSuffix(
+        'mobile-air-operations/v1/mobile-air-operations/page/check-in/',
+        input.reservation
+      ),
+      method: SwClient.Method.GET,
+      headers: input.headers
+    });
+  } catch (error) {
+    if (input.logger) {
+      const nowTimestamp = Luxon.DateTime.now().toLocaleString(
+        Luxon.DateTime.DATETIME_FULL_WITH_SECONDS
+      );
+      // It's normal to see many of these messages before we get the good response. We try
+      // many times because the airline's API has a quirk of not making checking available at
+      // exactly the time expected.
+      input.logger.log(
+        'Failed to fetch checkin data on attempt %d of %d at %s with error:',
+        input.attempt,
+        input.attemptLimit,
+        nowTimestamp,
+        error
+      );
     }
-  });
+  }
+
+  return response;
```

Comment:
See here, if we're making a bunch of individual request with the promises to delay when they actually run, we'll still return response regardless if it was a success or failed request... right?

Reply:
Great catch, I've updated the catch block to re-throw the error